great_expectations/core/datasource_dict.py

```diff
@@ -77,6 +89,10 @@ def __contains__(self, name: object) -> bool:
     def __setitem__(self, name: str, ds: FluentDatasource | BaseDatasource) -> None:
         config: FluentDatasource | DatasourceConfig
         if isinstance(ds, FluentDatasource):
+            if isinstance(ds, SupportsInMemoryDataAssets):
+                for asset in ds.assets:
+                    if asset.type == "dataframe":
```

Comment:
Do we have an enum for this?

Reply:
Not that I'm aware of. I'm not sure how we would keep it in sync with the data assets. We probably should defer that work.